src/librustdoc/html/markdown/tests.rs

```diff
@@ -86,23 +88,35 @@ fn test_lang_string_parse() {
     }
 
     // marker                | should_panic | no_run | ignore | rust | test_harness
-    //                       | compile_fail | allow_fail | error_codes | edition
-    t("", false, false, Ignore::None, true, false, false, false, v(), None);
-    t("rust", false, false, Ignore::None, true, false, false, false, v(), None);
-    t("sh", false, false, Ignore::None, false, false, false, false, v(), None);
-    t("ignore", false, false, Ignore::All, true, false, false, false, v(), None);
-    t("ignore-foo", false, false, ignore_foo, true, false, false, false, v(), None);
-    t("should_panic", true, false, Ignore::None, true, false, false, false, v(), None);
-    t("no_run", false, true, Ignore::None, true, false, false, false, v(), None);
-    t("test_harness", false, false, Ignore::None, true, true, false, false, v(), None);
-    t("compile_fail", false, true, Ignore::None, true, false, true, false, v(), None);
-    t("allow_fail", false, false, Ignore::None, true, false, false, true, v(), None);
-    t("{.no_run .example}", false, true, Ignore::None, true, false, false, false, v(), None);
-    t("{.sh .should_panic}", true, false, Ignore::None, false, false, false, false, v(), None);
-    t("{.example .rust}", false, false, Ignore::None, true, false, false, false, v(), None);
-    t("{.test_harness .rust}", false, false, Ignore::None, true, true, false, false, v(), None);
-    t("text, no_run", false, true, Ignore::None, false, false, false, false, v(), None);
-    t("text,no_run", false, true, Ignore::None, false, false, false, false, v(), None);
+    //                       | compile_fail | allow_fail | error_codes | edition | added classes
+    t("", false, false, Ignore::None, true, false, false, false, v(), None, v());
+    t("rust", false, false, Ignore::None, true, false, false, false, v(), None, v());
+    t("sh", false, false, Ignore::None, false, false, false, false, v(), None, v());
+    t("ignore", false, false, Ignore::All, true, false, false, false, v(), None, v());
+    t("ignore-foo", false, false, ignore_foo, true, false, false, false, v(), None, v());
+    t("should_panic", true, false, Ignore::None, true, false, false, false, v(), None, v());
+    t("no_run", false, true, Ignore::None, true, false, false, false, v(), None, v());
+    t("test_harness", false, false, Ignore::None, true, true, false, false, v(), None, v());
+    t("compile_fail", false, true, Ignore::None, true, false, true, false, v(), None, v());
+    t("allow_fail", false, false, Ignore::None, true, false, false, true, v(), None, v());
+    t("{.no_run .example}", false, true, Ignore::None, true, false, false, false, v(), None, v());
+    t("{.sh .should_panic}", true, false, Ignore::None, false, false, false, false, v(), None, v());
+    t("{.example .rust}", false, false, Ignore::None, true, false, false, false, v(), None, v());
```

Comment:
This is getting pretty unwieldy. Since you're modifying all the lines anyway, do you mind switching it to the builder pattern, so it's easier to read at a glance? Right now it's written for the machine to execute and only incidentally for a human to read; ideally it would be the opposite.

Reply:
Added an item to the todo-list for it, I'll do it later 😄

Reply:
Sounds good, yeah it doesn't need to block this new feature.